utils/aws/route53/__init__.py

```diff
@@ -0,0 +1,318 @@
+        """Returns the name of the account"""
+        return self._name
+
+    @property
+    def zones(self):
+        """
+        Returns the DNS zones managed under the account
+
+        :return: list of DNS zones
+        :rtype: list of Zone
+        """
+        return {a.name: a for a in self._zones}
+
+    def add_zone(self, zone):
+        """
+        Add a zone to the account
+
+        :param zone: DNS zone to add
+        :type zone: Zone
+        """
+        if self.zones.get(zone.name):
+            raise DuplicateException(zone, self)
+        self._zones.append(zone)
+
+    def get_zone(self, name):
+        """
+        Retrieve a zone by name
+
+        :param name: Zone name to retrieve
+        :type name: str
+        :return: DNS zone
+        :rtype: Zone
+        """
+        return self.zones.get(name)
+
+    def __repr__(self):
+        return f"Account<{self.name}>"
+
+
+class Zone(object):
+    """
+    Zone represents a DNS zone
+
+    :param name: DNS domain name
+    :param data: DNS zone data
+    :type name: str
+    :type data: dict
+    """
+    def __init__(self, name, data={}):
+        self._name = name.lower().rstrip('.')
+        self._records = []
+        self._data = data
+
+    @property
+    def data(self):
+        """Returns the zone data"""
+        return self._data
+
+    @property
+    def name(self):
+        """Returns the zone name"""
+        return self._name
+
+    @property
+    def records(self):
+        """Return DNS records"""
+        return {f"{r.name}": r for r in self._records}
+
+    def add_record(self, record):
+        """
+        Add a record to the zone
+
+        :param record: the record to add
+        :type record: Record
+        :raises DuplicateException: if a record by that name already exists
+        """
+        if self.records.get(record.name):
+            raise DuplicateException(record, self)
+        self._records.append(record)
+
+    def __eq__(self, other):
+        if not isinstance(other, Zone):
+            return False
+        return self.name == other.name
+
+    def __hash__(self):
+        return f"{self.name}".__hash__()
+
+    def __repr__(self):
+        return f"Zone<{self.name}>"
+
+
+class Record(object):
+    """
+    Record represents a DNS record
+
+    :param zone: the zone the record belongs to
+    :param name: the record name
+    :param data: the record data
+    :param awsdata: the original aws data for a dns record
+    :type zone: Zone
+    :type name: str
+    :type data: dict
+    :type awsdata: dict
+    """
+
+    SUPPORTED_TYPES = ['A', 'CNAME', 'NS', 'TXT']
+
+    def __init__(self, zone, name, data={}, awsdata={}):
+        self._zone = zone
+        self._name = _octal_replace(name).lower().rstrip('.')
+
+        try:
+            self._type = data['type']
+            self._ttl = int(data['ttl'])
+        except KeyError as e:
+            raise InvalidRecordData(f"missing key {e} in Record data")
```

Comment:
I love that you created custom exceptions. For correctness, the `InvalidRecordData` should inherit from AttributeError, but that's just me being super picky. That would allow callers to handle class initialization exceptions without having to import `InvalidRecordData`.

Reply:
fixed in c75921417e8733e6524d8470d2aef7a8fe1547c2